Add release workflows

theory opened the pull request:
Follow the example of houseabsolute/ubi to use cross-rs/cross to cross-compile binaries for multiple platforms and architectures. Use `musl` where available to support both `glibc` and `musl` Linux distributions. Test for every push against stable, beta, and nightly, and build archives that contain the license, readme, and change log, as well as the `pgxn_meta` binary.

Also add a workflow to publish on crates.io, and add a dependency check step to the lint and test workflow.

Add `CHANGELOG.md` for tracking changes, and update `Cargo.toml` for distribution on crates.io.

Revise the README for the crate and binary.

ChuckHend commented on README.md:
````diff
@@ -51,6 +53,55 @@ func main() {
+
+There are several ways to install `pgxn_meta`.
+
+### Use `ubi`
+
+Install the [universal binary installer (ubi)](https://github.com/houseabsolute/ubi) and
+use it to install `pgxn_meta` and many other tools.
+
+``` console
+$ ubi --project pgxn/meta --in ~/bin
````
Curious when/why we would use `ubi` instead of just `cargo`? I'm asking out of ignorance because I have never used ubi.

theory replied:
Oh I meant to include Cargo instructions too. `ubi` installs binaries from GitHub, so it's fast.

theory replied:
Oh, the Cargo instructions are below. Phew!

theory committed:
Add release workflows

Follow the example of houseabsolute/ubi to use cross-rs/cross to
cross-compile binaries for multiple platforms and architectures. Use
`musl` where available to support both `glibc` and `musl` Linux
distributions. Test for every push against stable, beta, and nightly,
and build archives that contain the license, readme, and change log, as
well as the `pgxn_meta` binary.

Also add a workflow to publish on crates.io, and add a dependency check
step to the lint and test workflow.

Add `CHANGELOG.md` for tracking changes, and update `Cargo.toml` for
distribution on crates.io.

Revise the README for the crate and binary.

vrmiguel reviewed (approved):
Looks great! I like the emojis :v 

🐧 😈
